### PR TITLE
Add openshift/sre-ssh-proxy-container repo

### PR DESCRIPTION
**Pull request by mbarnes**
For https://issues.redhat.com/browse/OSD-5071

### CODE REVIEW

File: core-services/prow/02_config/_plugins.yaml [hunk 1]
```diff
@@ -2604,6 +2608,9 @@ lgtm:
   - openshift-priv/cluster-baremetal-operator
   - openshift/cluster-baremetal-operator
   review_acts_as_lgtm: true
+- repos:
+  - openshift/sre-ssh-proxy-container
+  review_acts_as_lgtm: true
```
Comment (cblecker) [inline core-services/prow/02_config/_plugins.yaml:2613]: This isn't needed

File: core-services/prow/02_config/_plugins.yaml [hunk 2]
```diff
@@ -299,6 +299,10 @@ approve:
   - openshift-priv/cluster-baremetal-operator
   - openshift/cluster-baremetal-operator
   require_self_approval: false
+- commandHelpLink: ""
+  repos:
+  - openshift/sre-ssh-proxy-container
+  require_self_approval: false
```
Comment (cblecker) [inline core-services/prow/02_config/_plugins.yaml:305]: This isn't needed

File: ci-operator/config/openshift/sre-ssh-proxy-container/openshift-sre-ssh-proxy-container-master.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tag_specification:
+  name: "4.6"
+  namespace: ocp
```
Comment (cblecker) [inline ci-operator/config/openshift/sre-ssh-proxy-container/openshift-sre-ssh-proxy-container-master.yaml:23]: We aren't distributing as a part of OCP, so this isn't needed.
```suggestion
```

File: ci-operator/config/openshift/sre-ssh-proxy-container/openshift-sre-ssh-proxy-container-master.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    namespace: openshift
+    tag: "7"
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.13
+promotion:
+  name: "4.6"
+  namespace: ocp
```
Comment (cblecker) [inline ci-operator/config/openshift/sre-ssh-proxy-container/openshift-sre-ssh-proxy-container-master.yaml:13]: We aren't distributing as a part of OCP, so this isn't needed.
```suggestion
```

File: ci-operator/config/openshift/sre-ssh-proxy-container/openshift-sre-ssh-proxy-container-master.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+base_images:
+  os:
+    name: centos
+    namespace: openshift
+    tag: "7"
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.13
```
Comment (cblecker) [inline ci-operator/config/openshift/sre-ssh-proxy-container/openshift-sre-ssh-proxy-container-master.yaml:10]:
```suggestion
build_root:
  image_stream_tag:
    name: release
    namespace: openshift
    tag: golang-1.13
images:
- dockerfile_path: container/Dockerfile
  from: base
  to: sre-ssh-proxy-container
```

File: ci-operator/config/openshift/sre-ssh-proxy-container/openshift-sre-ssh-proxy-container-master.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      cpu: 100m
+      memory: 200Mi
+tag_specification:
+  name: "4.6"
+  namespace: ocp
+tests:
+- as: make
+  commands: make
+  container:
+    from: src
```
Comment (cblecker) [inline ci-operator/config/openshift/sre-ssh-proxy-container/openshift-sre-ssh-proxy-container-master.yaml:28]: We don't have access to the raw daemon socket, so we need to use image stream builds in CI
```suggestion
```

**Review by cblecker — CHANGES_REQUESTED** [review]

**Comment by mbarnes** [comment]
Applied all suggestions and squashed.

**Comment by cblecker** [comment]
@mbarnes Can you run `make jobs` to regenerate the config, and also copy https://raw.githubusercontent.com/openshift/sre-ssh-proxy-container/master/OWNERS into `ci-operator/config/openshift/sre-ssh-proxy-container/` and `ci-operator/jobs/openshift/sre-ssh-proxy-container/` ?

**Comment by mbarnes** [comment]
> @mbarnes Can you run make jobs to regenerate the config, and also copy https://raw.githubusercontent.com/openshift/sre-ssh-proxy-container/master/OWNERS into ci-operator/config/openshift/sre-ssh-proxy-container/ and ci-operator/jobs/openshift/sre-ssh-proxy-container/ ?

Done; amended the commit.

File: ci-operator/config/openshift/sre-ssh-proxy-container/openshift-sre-ssh-proxy-container-master.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+base_images:
+  os:
```
Comment (cblecker) [inline ci-operator/config/openshift/sre-ssh-proxy-container/openshift-sre-ssh-proxy-container-master.yaml:2]:
```suggestion
  base:
```

**Comment by mbarnes** [comment]
Appears to be using the wrong build context.

`make` runs `"$(CONTAINER_ENGINE) build container"` where `CONTAINER_ENGINE` is `docker` or `podman`.

Works fine locally with Docker, but CI job is failing trying to find `container/pam_sshd`.

**Comment by cblecker** [comment]
/assign @alvaroaleman 

This should be good. We can fix the rehearse in-repo.

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**